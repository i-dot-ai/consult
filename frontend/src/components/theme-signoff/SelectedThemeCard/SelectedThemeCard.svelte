<script lang="ts">
  import clsx from "clsx";

  import { fade, fly } from "svelte/transition";

  import type { GeneratedTheme } from "../../../global/types";

  import Panel from "../../dashboard/Panel/Panel.svelte";
  import Button from "../../inputs/Button/Button.svelte";
  import MaterialIcon from "../../MaterialIcon.svelte";
  import Delete from "../../svg/material/Delete.svelte";
  import Docs from "../../svg/material/Docs.svelte";
  import EditSquare from "../../svg/material/EditSquare.svelte";
  import AddCustomTheme from "../AddCustomTheme/AddCustomTheme.svelte";
  import AnswersList from "../AnswersList/AnswersList.svelte";

  export interface Props {
    theme: GeneratedTheme,
    answers: string[],
    removeTheme: (themeId: string) => void,
    updateTheme: (themeId: string, title: string, description: string) => void,
  }

  let {
    theme,
    answers = [],
    removeTheme = () => {},
    updateTheme = () => {},
  }: Props = $props();

  let showAnswers = $state(false);
  let editing = $state(false);
</script>

<article class="bg-white rounded-lg">
  {#if editing}
    <div in:fade>
      <AddCustomTheme
        variant="edit"
        handleCancel={() => (editing = false)}
        handleConfirm={(title, description) => {
          updateTheme(theme.id, title, description);
        }}
      />
    </div>
  {:else}
    <div in:fade>
      <Panel>
        <div class="grid grid-cols-12">
          <div class={clsx([showAnswers ? "col-span-4" : "col-span-12"])}>
            <header>
              <h2>{theme.name}</h2>
            </header>

            <p class="text-sm text-neutral-700 my-4">
              {theme.description}
            </p>

            <footer class="flex items-center flex-wrap gap-2">
              <Button size="sm" handleClick={() => (editing = !editing)}>
                <MaterialIcon color="fill-neutral-500">
                  <EditSquare />
                </MaterialIcon>
                Edit
              </Button>

              <Button size="sm" handleClick={() => removeTheme(theme.id)}>
                <MaterialIcon color="fill-neutral-500">
                  <Delete />
                </MaterialIcon>
                Remove
              </Button>

              <Button
                size="sm"
                handleClick={() => (showAnswers = !showAnswers)}
                disabled={!answers.length}
              >
                <MaterialIcon color="fill-neutral-500">
                  <Docs />
                </MaterialIcon>
                <span class="block w-full text-start">
                  Respresentative Responses
                </span>
              </Button>
            </footer>
          </div>

          {#if showAnswers}
            <aside
              in:fly={{ x: 300 }}
              class="col-span-8 border-l border-neutral-200 ml-4 pl-4"
            >
              <AnswersList title="Representative Responses" {answers} />
            </aside>
          {/if}
        </div>
      </Panel>
    </div>
  {/if}
</article>
