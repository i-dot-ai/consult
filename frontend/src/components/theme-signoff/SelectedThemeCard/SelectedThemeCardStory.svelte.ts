import SelectedThemeCard from "./SelectedThemeCard.svelte";

let theme = $state({
  id: "theme-id",
  name: "Theme Name",
  description: "Theme description",
});
let answers = $state(["Answer 1", "Answer 2"]);
let removeTheme = $state((themeId) =>
  alert(`Remove theme event triggered with: ${themeId}`),
);
let updateTheme = $state((...args) =>
  alert(`Update theme event triggered with: ${args.join(", ")}`),
);

export default {
  name: "SelectedThemeCard",
  component: SelectedThemeCard,
  category: "Theme Signoff",
  props: [
    {
      name: "theme",
      value: theme,
      type: "json",
    },
    {
      name: "answers",
      value: answers,
      type: "json",
    },
    {
      name: "removeTheme",
      value: removeTheme,
      type: "func",
      schema: `(themeId: string) => void`,
    },
    {
      name: "updateTheme",
      value: updateTheme,
      type: "func",
      schema: `(themeId: string, title: string, description: string) => void,`,
    },
  ],
  stories: [],
};
