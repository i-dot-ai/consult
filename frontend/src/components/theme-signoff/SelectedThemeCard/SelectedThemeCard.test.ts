import { afterEach, beforeEach, describe, expect, it } from "vitest";
import { render, cleanup } from "@testing-library/svelte";

import SelectedThemeCard, { type Props } from "./SelectedThemeCard.svelte";

let testData: Props;

describe("SelectedThemeCard", () => {
  beforeEach(() => {
    testData = {
      theme: {
        id: "theme-id",
        name: "Theme Name",
        description: "This is the theme description",
      },
      answers: ["Answer 1", "Answer 2"],
      removeTheme: () => {},
      updateTheme: () => {},
    };
  });

  afterEach(() => cleanup());

  it("should render", async () => {
    const { container, getByText, queryByText } = render(SelectedThemeCard, {
      ...testData,
    });

    expect(getByText(testData.theme.name));
    expect(getByText(testData.theme.description));

    // Answers hidden initially
    testData.answers?.forEach((answer) => {
      expect(queryByText(answer)).toBeNull();
    });

    expect(container).toMatchSnapshot();
  });
});
