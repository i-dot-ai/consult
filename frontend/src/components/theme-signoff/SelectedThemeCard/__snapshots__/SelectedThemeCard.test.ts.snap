// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelectedThemeCard > should render 1`] = `
<div>
  <article
    class="bg-white rounded-lg"
  >
    <div>
      <div
        class="h-full my-4 p-4 rounded-2xl border border-gray-200"
        data-testid="panel-component"
      >
        <div
          class="grid grid-cols-12"
        >
          <div
            class="col-span-12"
          >
            <header>
              <h2>
                Theme Name
              </h2>
            </header>
             
            <p
              class="text-sm text-neutral-700 my-4"
            >
              This is the theme description
            </p>
             
            <footer
              class="flex items-center flex-wrap gap-2"
            >
              <!---->
              <button
                aria-pressed="false"
                class="text-sm cursor-pointer rounded-md py-1 px-2 border bg-white border-gray-300 transition-colors duration-300 self-start inline-flex gap-1 items-center hover:bg-gray-100 svelte-1rjb6fa"
                data-variant="default"
                role="button"
                tabindex="0"
              >
                <!---->
                <!---->
                <svg
                  class="fill-neutral-500"
                  height="1rem"
                  role="img"
                  viewBox="0 -960 960 960"
                  width="1rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M200-120q-33 0-56.5-23.5T120-200v-560q0-33 23.5-56.5T200-840h357l-80 80H200v560h560v-278l80-80v358q0 33-23.5 56.5T760-120H200Zm280-360ZM360-360v-170l367-367q12-12 27-18t30-6q16 0 30.5 6t26.5 18l56 57q11 12 17 26.5t6 29.5q0 15-5.5 29.5T897-728L530-360H360Zm481-424-56-56 56 56ZM440-440h56l232-232-28-28-29-28-231 231v57Zm260-260-29-28 29 28 28 28-28-28Z"
                  />
                  <!---->
                </svg>
                <!---->
                 Edit
                
                
              </button>
              
              <!---->
               
              <!---->
              <button
                aria-pressed="false"
                class="text-sm cursor-pointer rounded-md py-1 px-2 border bg-white border-gray-300 transition-colors duration-300 self-start inline-flex gap-1 items-center hover:bg-gray-100 svelte-1rjb6fa"
                data-variant="default"
                role="button"
                tabindex="0"
              >
                <!---->
                <!---->
                <svg
                  class="fill-neutral-500"
                  height="1rem"
                  role="img"
                  viewBox="0 -960 960 960"
                  width="1rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M280-120q-33 0-56.5-23.5T200-200v-520h-40v-80h200v-40h240v40h200v80h-40v520q0 33-23.5 56.5T680-120H280Zm400-600H280v520h400v-520ZM360-280h80v-360h-80v360Zm160 0h80v-360h-80v360ZM280-720v520-520Z"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                  <!---->
                </svg>
                <!---->
                 Remove
                
                
              </button>
              
              <!---->
               
              <!---->
              <button
                aria-pressed="false"
                class="text-sm cursor-pointer rounded-md py-1 px-2 border bg-white border-gray-300 transition-colors duration-300 self-start inline-flex gap-1 items-center hover:bg-gray-100 svelte-1rjb6fa"
                data-variant="default"
                role="button"
                tabindex="0"
              >
                <!---->
                <!---->
                <svg
                  class="fill-neutral-500"
                  height="1rem"
                  role="img"
                  viewBox="0 -960 960 960"
                  width="1rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M320-440h320v-80H320v80Zm0 120h320v-80H320v80Zm0 120h200v-80H320v80ZM240-80q-33 0-56.5-23.5T160-160v-640q0-33 23.5-56.5T240-880h320l240 240v480q0 33-23.5 56.5T720-80H240Zm280-520v-200H240v640h480v-440H520ZM240-800v200-200 640-640Z"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                  <!---->
                </svg>
                <!---->
                 
                <span
                  class="block w-full text-start"
                >
                  Respresentative Responses
                </span>
                
                
              </button>
              
              <!---->
            </footer>
          </div>
           
          <!---->
        </div>
        <!---->
      </div>
      <!---->
    </div>
    <!---->
  </article>
  
</div>
`;
